### ISSUE TITLE
add `reportUnusedParameter` rule

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#336]
```py
def foo(asdf: int): # no error when asdf is unused
    pass
```
this was probably caused by #326

**Comment by DetachHead** [comment]
looks like this is because `reportUnusedVariable` doesn't report error/warning/info diagnostic levels on function parameters, but it always reported the "unused" diagnostic. i think for backwards compatibility we should fix this by making a separate `reportUnusedParameter` rule

**Comment by vikigenius** [comment]
What's the status on this? Currently I have no way of specifying how to handle unused function parameters.

**Comment by DetachHead** [comment]
can you add a 👍 react on this issue to vote for it. helps us prioritise based on demand. thanks!
